conan/tools/microsoft/msbuilddeps.py

```diff
@@ -84,6 +88,10 @@ def generate(self):
         # config_filename
         # TODO: This is duplicated in write_generators() function, would need to be moved
         # to generators and called from there
+        exclude_code_analysis = self._conanfile.conf["tools.microsoft.msbuilddeps"].exclude_code_analysis
+        if exclude_code_analysis is not None:
+                self.exclude_code_analysis = eval(exclude_code_analysis)
+                assert isinstance(self.exclude_code_analysis, (bool, list))
```

Comment:
@memsharded I've applied your suggestions, but now, string pattern doesn't work any longer, as you accept only bool and list here.